tasks.py.jinja

```diff
@@ -309,6 +354,115 @@ def start(c, detach=True, debugpy=False):
+    """Install Odoo addons"""
+    if not (modules or core or extra or private):
+        raise exceptions.ParseError(
+            message="You must provide at least one option. "
+            "See --help for details."
+        )
+    cmd = "docker-compose run --rm addons init"
+    if core:
+        cmd += " --core"
+    if extra:
+        cmd += " --extra"
+    if private:
+        cmd += " --private"
+    if modules:
+        cmd += f" -w {modules}"
+    with c.cd(str(PROJECT_ROOT)):
+        c.run(
+            cmd,
+            env=dict(
+                UID_ENV,
+            ),
+        )
+
+
+def _get_cwd_addon(file=None):
+    if not file:
+        cwd = Path.cwd()
+    else:
+        cwd = Path.parent(file)
+    manifest_file = False
+    while not manifest_file:
+        manifest_file = (
+            Path(cwd / "__manifest__.py").exists()
+            or Path(cwd / "__openerp__.py").exists()
+        )
+        if manifest_file:
+            return cwd.stem
+        cwd = cwd.parent
+        if cwd == PROJECT_ROOT:
+            return None
+
+
+@task(
+    develop,
+    help={
+        "debugpy": "Whether or not to run tests in a VSCode debugging session. "
+        "Default: False",
+        "modules": "Comma-separated list of modules to test. Default: 'base'.",
+        "logs": "Whether or not to follow Odoo logs during testing. " "Default: True",
+    },
+)
+def test(c, modules=None, debugpy=False, logs=True, cur_file=None):
+    """Run Odoo tests
+    NOTE: Odoo must be restarted manually after this to go back to normal mode
+    """
+    if not modules:
+        cur_module = _get_cwd_addon(file=cur_file)
+        modules = cur_module if cur_module is not None else "base"
+    tmp_docker_compose_file = tempfile.NamedTemporaryFile(
+        dir=PROJECT_ROOT, mode="w", suffix=".yaml"
+    )
+    cmd = (
+        "docker-compose -f docker-compose.yml "
+        f"-f {tmp_docker_compose_file.name} up -d"
+    )
+    odoo_command = f"addons init -tw {modules}"
+    _override_docker_command(c, "odoo", odoo_command, file=tmp_docker_compose_file)
+    with c.cd(str(PROJECT_ROOT)):
+        c.run(
+            cmd,
+            env=dict(
+                UID_ENV,
+                DOODBA_DEBUGPY_ENABLE=str(int(debugpy)),
+            ),
+        )
+        if logs:
+            c.run(
+                "docker-compose logs -f odoo",
+                env=dict(
+                    UID_ENV,
+                ),
+            )
+    _logger.info("Waiting for services to spin up...")
+    time.sleep(SERVICES_WAIT_TIME)
+    tmp_docker_compose_file.close()
+
+
+def _override_docker_command(c, service, command, file=None):
+    docker_config = {
+        "version": "2.4",  # TODO: Update this if main file is updated
+        "services": {service: {"command": command.split(" ")}},
+    }
+    docker_config_yaml = yaml.dump(docker_config)
+    if file:
+        file.write(docker_config_yaml)
+        file.flush()
```

Comment:
Is this function useful for something without `file`? :thinking:

Reply:
No, indeed. I was thinking of other uses when I wrote it but it ended up only being useful with an output file. Fixed. TY!